[SYCL][Matrix]Add test for all sizes

YuriPlyakhin opened the pull request.

dkhaldi committed:
[SYCL][Matrix]Add test for odd sizes

dkhaldi committed:
set to xfail on GPU

dkhaldi committed:
Uncomment the rest of the calls

dkhaldi committed:
correct the verification code

YuriPlyakhin committed:
fixed M size of the big matrix to be multiplier of M size of joint matrix

YuriPlyakhin committed:
Merge branch 'intel:intel' into all_sizes_tst

YuriPlyakhin committed:
[SYCL][Matrix][NFC] updated formatting

Updated formatting.

dkhaldi reviewed (commented):
LGTM but we should add the XMX8  version of the test so it gets tested on ATS-M as well

dkhaldi commented on SYCL/Matrix/joint_matrix_all_sizes.cpp:
```diff
@@ -0,0 +1,25 @@
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// XFAIL: gpu
```
do we keep xfail?

YuriPlyakhin replied:
yes, while the problem with multiple kernels in the same test is not resolved.

YuriPlyakhin committed:
[SYCL][Matrix] Updated tests to pass on GPU

Updated tests to pass on GPU, splitting compilation of each kernel.
Added new environment variables for different optimization modes
Added XMX8 variant.

YuriPlyakhin committed:
Merge branch 'intel:intel' into all_sizes_tst

dkhaldi reviewed (changes requested):
I prefer the way it was with having multiple tests rather than macros. what do you think?
It is okay to set it to xfail until we fix the bug

dkhaldi commented on SYCL/Matrix/joint_matrix_all_sizes_impl.hpp:
```diff
@@ -0,0 +1,230 @@
+
+int main() {
+#if defined(JOINT_MATRIX_TEST_BF16_1) || defined(JOINT_MATRIX_TEST_ALL_BF16)
+  init_and_multiply<bfloat16, float, 2, 1, 16, 16>();
```
change N from 16 to SG_SZ

YuriPlyakhin replied:
resolved.

YuriPlyakhin committed:
[SYCL][Matrix] Fixed size N - it should be different depending on architecture

dkhaldi reviewed (approved):
LGTM
